### PR TITLE
Fix API connection by moving types in a different file

### PR DESCRIPTION
**Pull request by KarishmaBothara**
I am not able to produce the issue with API connection with this repo.. But when I try to install the latest api and use the doc example - docs/examples/promise/01_simple_connect/index.js it fails with - 
```
node docs/examples/promise/01_simple_connect/index.js 
Unknown signed extensions [OptionSignedExtension] found, treating them as no-efect
Unknown signed extensions [OptionSignedExtension] found, treating them as no-efect
Unknown types found, no types for LastRuntimeUpgradeInfo,RewardBalanceOf,schnorrkel::Randomness,schnorrkel::RawVRFOutput
Unknown signed extensions [OptionSignedExtension] found, treating them as no-efect
Unknown signed extensions [OptionSignedExtension] found, treating them as no-efect
2020-08-17 15:05:07   API/DECORATOR: Error: FATAL: Unable to initialize the API: section[methodName] is not a function
    at EventEmitter.Init._onProviderConnect (/Users/karishma/WebstormProjects/node_modules/@polkadot/api/base/Init.js:53:23)
    at process._tickCallback (internal/process/next_tick.js:68:7)
Error: Connection fail
    at EventEmitter.rejectError (/Users/karishma/WebstormProjects/node_modules/@cennznet/api/Api.js:42:16)
    at EventEmitter.emit (/Users/karishma/WebstormProjects/node_modules/eventemitter3/index.js:202:33)
    at Api.emit (/Users/karishma/WebstormProjects/node_modules/@polkadot/api/base/Events.js:21:31)
    at EventEmitter.Init._onProviderConnect (/Users/karishma/WebstormProjects/node_modules/@polkadot/api/base/Init.js:55:14)
    at process._tickCallback (internal/process/next_tick.js:68:7)

```

Still looking at ways to produce this issue in the API.. but the work around is to move types in a separate file.

### CODE REVIEW

File: packages/api/src/derives/types.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import {RewardDestination} from '@cennznet/types';
+import {AccountId, EraIndex, Exposure, Keys, StakingLedger, ValidatorPrefs} from '@polkadot/types/interfaces';
+
+export interface DerivedStakingInfo {
```
Comment (benxgao) [inline packages/api/src/derives/types.ts:4]: Do we always group all types files inside types folder, or any difference on this one?

Comment (KarishmaBothara) [reply packages/api/src/derives/types.ts:4]: Its good to group all types in a file cause else api think its part of its derive

File: packages/api/src/derives/types.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import {RewardDestination} from '@cennznet/types';
+import {AccountId, EraIndex, Exposure, Keys, StakingLedger, ValidatorPrefs} from '@polkadot/types/interfaces';
```
[the remark this answers is not in the document]
Comment (KarishmaBothara) [reply packages/api/src/derives/types.ts]: done

File: packages/api/src/derives/staking/stakingAccount.ts [hunk 3]
```diff
@@ -60,7 +48,7 @@ export function queryStakingAccountInfo(
 
       return retrieveStakingAccountDetails(api, stashId).pipe(
         switchMap(
-          ([controllerIdOpt, nominatorsOpt, rewardDestination, stakers, [validatorPrefs], nextKeys]): Observable<
+          ([controllerIdOpt, nominatorsOpt, rewardDestination, stakers, [validatorPrefs]]): Observable<
```
Comment (alexsednz) [inline packages/api/src/derives/staking/stakingAccount.ts:51]: Can you please explain what is this change for?

Comment (KarishmaBothara) [reply packages/api/src/derives/staking/stakingAccount.ts:51]: the nextKeys does not exist.. 
The query for four fields  -  [api.query.staking.bonded, stashId],
    [api.query.staking.nominators, stashId],
    [api.query.staking.payee, stashId],
    [api.query.staking.stakers, stashId],
    [api.query.staking.validators, stashId], are mapped to 
[controllerIdOpt, nominatorsOpt, rewardDestination, stakers, [validatorPrefs]]

Comment (alexsednz) [reply packages/api/src/derives/staking/stakingAccount.ts:51]: The question is would it make anything fail? If so, shouldn't we add a test to catch such an issue later?

**Pull request merged**